src/app/server-cluster/ServerClusterInterface.h

```diff
@@ -0,0 +1,198 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/AttributeValueDecoder.h>
+#include <app/AttributeValueEncoder.h>
+#include <app/CommandHandler.h>
+#include <app/ConcreteClusterPath.h>
+#include <app/data-model-provider/ActionReturnStatus.h>
+#include <app/data-model-provider/MetadataList.h>
+#include <app/data-model-provider/MetadataTypes.h>
+#include <app/data-model-provider/OperationTypes.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/support/BitFlags.h>
+
+namespace chip {
+namespace app {
+
+/// Defines an active cluster on an endpoint.
+///
+/// Provides metadata as well as interaction processing (attribute read/write and command handling).
```

Comment:
Suggest mentioning that it has internal intrusive list semantics.

Reply:
I was torn here ... I want the "intrusive list" to be an implementation detail and *not* a stable API. For efficiency we could opt for example on https://pigweed.dev/pw_containers/maps.html#pw-intrusivemap  (looks neat to get log search guaranteed, unsure about overhead).

Would you prefer a mention of "this has intrusive list" with stating that the API there is NOT stable? I tried to add those to every method, but can also split it in a separate class and have it noted that the dependency there is not stable API.